pkg/agent/agent.go

```diff
@@ -77,15 +75,22 @@ func (a *Agent) run() error {
 		}
 	}
 
-	ctx, err := a.startManager(svid, key, bundle)
+	err = a.startManager(svid, key, bundle)
 	if err != nil {
 		return err
 	}
 
-	a.t.Go(func() error { return a.managerWait(ctx) })
 	a.t.Go(func() error { return a.startEndpoints(bundle) })
 
-	<-a.t.Dying()
+	// Wait until the agent's tomb is dying or the manager stopped working.
+	select {
+	case <-a.t.Dying():
+	case <-a.Manager.Stopped():
+		a.mtx.Lock()
+		// TODO: Should we try to restart manager here?
```

Comment:
Probably :) This might be easier and cleaner if done in a dedicated function like `a.t.Go(a.superviseManager)` where we can implement logic to try to restart if necessary